node/flatpak-node-generator.py

```diff
@@ -619,11 +652,51 @@ def process_lockfile(self, lockfile: Path) -> Iterator[Package]:
         raise NotImplementedError()
 
 
+class RCFileProvider:
+    def get_node_headers(self, rcfile: Path) -> NodeHeaders:
+        raise NotImplementedError()
+
+
 class ModuleProvider(contextlib.AbstractContextManager):
     async def generate_package(self, package: Package) -> None:
         raise NotImplementedError()
 
 
+class NodeHeadersProvider(contextlib.AbstractContextManager):
+    def __init__(self, gen: ManifestGenerator):
+        self.gen = gen
+
+    def __exit__(self, *_: Any) -> None:
+        pass
+
+    def get_gyp_dir(self) -> Path:
+        return self.gen.data_root / 'cache' / 'node-gyp'
+
+    async def generate_node_headers(self, node_headers: NodeHeaders, dest: Path = None):
+        url = node_headers.get_url()
+        install_version = node_headers.get_install_version()
+        if dest is None:
+            dest = self.get_gyp_dir() / node_headers.target
+        metadata = await RemoteUrlMetadata.get(url, cachable=True)
+        self.gen.add_archive_source(url,
+                                    metadata.integrity,
+                                    destination=dest)
+        self.gen.add_data_source(install_version,
+                                 destination=dest / 'installVersion')
+
+    async def generate_sdk_node_headers(self):
+        gyp_dir = self.get_gyp_dir()
+        install_version = "9"
+        script = f'''version="$(node --version | sed \'s/^v//\')"
+                     nodedir="$(dirname $(dirname $(which node)))"
+                     mkdir -p "{gyp_dir}/$version"
+                     ln -s "$nodedir/include" "{gyp_dir}/$version/include"
+                     echo "{install_version}" > "{gyp_dir}/$version/installVersion"'''
+        self.gen.add_command('\n'.join(l.strip() for l in script.splitlines()))
```

Comment:
Would it? It'll be even longer and IMO just as hard to read.

Reply:
Well that's pretty much what the function is in the stdlib for... Actually in this case you wouldn't even need the `lstrip`, it would *just* be `self.gen.add_command(textwrap.dedent(script))`.